src/ol/interaction/selectinteraction.js

```diff
@@ -150,7 +166,10 @@ ol.interaction.Select.handleEvent = function(mapBrowserEvent) {
          * @param {ol.layer.Layer} layer Layer.
          */
         function(feature, layer) {
-          selected.push(feature);
+          if (!goog.isDef(filter) || goog.isFunction(filter) &&
+              filter.call(this, feature)) {
```

Comment:
Note that we do call the `layerFilter` with the interaction as `this` so I don't know why we'd do it for the `filter`.